Documentation/AEPAudience.md

```diff
@@ -0,0 +1,223 @@
+  [AEPMobileCore configureWithAppId:@"appId"];
+   }];
+   return YES;
+}
+```
+
+
+
+
+# Audience API reference
+
+This section details all the APIs provided by AEPAudience, along with sample code snippets on how to properly use the APIs.
+
+## extensionVersion
+
+The `extensionVersion()` API returns the version of the Audience extension that is registered with the Mobile Core extension.
+
+**Examples**
+
+**Swift**
+
+```
+let version = Audience.extensionVersion
+```
+
+**Objective-C**
+
+```
+NSString *version = [AEPMobileAudience extensionVersion];
+```
+
+
+
+## getVisitorProfile
+
+Returns the visitor profile that was most recently updated. The visitor profile is saved in the SDK's local storage for access across multiple launches of your app. If no audience signal has been sent before, when this API is called, a null value is returned.
+
+**Syntax**
+
+```swift
+static func getVisitorProfile(completion: @escaping ([String: String]?, Error?) -> Void)
+```
+
+**Examples**
+
+**Swift**
+
+```swift
+Audience.getVisitorProfile { (visitorProfile, error) in
+                 print("#getVisitorProfile - retrieved profile: \(String(describing: visitorProfile))")
+                 if error != nil {
+                   // handle the error here
+                   }
+                  }
+```
+
+**Objective-C**
+
+```objectivec
+[AEPMobileAudience getVisitorProfile:^(NSDictionary<NSString *,NSString *> * _Nullable visitorProfile, NSError * _Nullable error) {
+            if (error) {
+                // handle the error here
+              } else {
+                // handle the returned visitorProfile dictionary here
+              }
+        }];
+```
+
+
+
+## signalWithData
+
+Use this method to send a signal with traits to Audience Manager and get the matching segments for the visitor in a closure. Audience manager sends the UUID in response to an initial signal call. The UUID is persisted on local SDK storage and is sent by the SDK to Audience Manager in all subsequent signal requests.
+
+If you are using the Experience Cloud ID \(ECID\) Service \(formerly MCID\), the ECID and other custom identifiers for the same visitor are sent with each signal request. The visitor profile that is returned by Audience Manager is saved in SDK local storage and is updated with subsequent signal calls.
+
+ℹ️ For more information about the UUID and other Audience Manager identifiers, see [Index of IDs in Audience Manager](https://marketing.adobe.com/resources/help/en_US/aam/ids-in-aam.html).
+
+
+**Syntax**
+
+```swift
+static func signalWithData(data: [String: String], completion: @escaping ([String: String]?, Error?) -> Void) 
+```
+
+**Examples**
+
+**Swift**
+
+```swift
+Audience.signalWithData(data: ["trait": "trait value"]) { (traits, error) in
+            // handle the returned visitorProfile dictionary here
+            if error != nil {
+            // handle the error here
+                 }
```

Comment:
Let's adjust the indentation and make it uniform. Also update the sample to match Obj-C Sample.